src/coreclr/jit/optimizebools.cpp

```diff
@@ -1576,11 +1576,13 @@ PhaseStatus Compiler::optOptimizeBools()
         printf("*************** In optOptimizeBools()\n");
     }
 #endif
-    bool     change    = false;
-    bool     retry     = false;
-    unsigned numCond   = 0;
-    unsigned numPasses = 0;
-    unsigned stress    = false;
+    bool          change     = false;
+    bool          retry      = false;
+    unsigned      numCond    = 0;
+    unsigned      numPasses  = 0;
+    unsigned      stress     = false;
+    BitVecTraits* ccmpTraits = new BitVecTraits(fgBBNumMax + 1, this);
```

Comment:
```suggestion
    BitVecTraits ccmpTraits(fgBBNumMax + 1, this);
```
These can all be like this, this existing code allocates and leaks heap memory.